src/poetry/puzzle/provider.py

```diff
@@ -352,11 +353,11 @@ def search_for_url(self, dependency: URLDependency) -> List[Package]:
     @classmethod
     def get_package_from_url(cls, url: str) -> Package:
         with temporary_directory() as temp_dir:
-            temp_dir = Path(temp_dir)
+            _temp_dir = Path(temp_dir)
```

Comment:
Slightly unrelated to the original goal here, but I wouldn't mind seeing this switched to stdlib `TemporaryDirectory` in this PR, which should also alleviate the need for the `_temp_dir` shuffle.

Reply:
done. note `TemporaryDirectory` still returns a `str`, but refactored a bit to remove the unnecessary intermediate var